src/Core/GoogleSheetsStorage.js

```diff
@@ -355,10 +355,44 @@ var GoogleSheetsStorage = class GoogleSheetsStorage extends AbstractStorage {
      * @param columnIndex (integer) optional; column index
      */
     addColumn(columnName, columnIndex = 1) {
-    
+
       // header of the column at the desider index is not empty
-      if( this.SHEET.getRange(`A${columnIndex}`).getValue() != "" ) {
-        this.SHEET.insertColumns(columnIndex);
+      try {
+        if( this.SHEET.getRange(`A${columnIndex}`).getValue() != "" ) {
+          this.SHEET.insertColumns(columnIndex);
+        }
+      } catch (e) {
```

Comment:
@zapolsky please check if possible to replace the try/catch fallback with explicit pre-checks: first validate the columnIndex against this.SHEET.getMaxColumns(), then use this.SHEET.getRange(1, columnIndex) to check if the header cell is empty and extract the loop that finds the first empty column into a dedicated helper function. This removes reliance on exceptions for normal control flow